City-of-Helsinki/berth-reservations-admin
# VEN-722, VEN-723, VEN-724 | Implement winter storage applications pages

## Description :sparkles:

This PR contains the initial implementation of the winter storage applications listing page as well as the individual application page.

### Closes :no_good_woman:

[VEN-722](https://helsinkisolutionoffice.atlassian.net/browse/VEN-722)
[VEN-723](https://helsinkisolutionoffice.atlassian.net/browse/VEN-723)
[VEN-724](https://helsinkisolutionoffice.atlassian.net/browse/VEN-724)

## Testing :alembic:

### Automated tests :gear:️

- Added happy path browser test in `browser-tests/navigateAndView.ts` that includes the navigation flow to both pages.
- `src/features/winterStorageApplicationList/__tests__/utils.test.ts`
- `src/features/winterStorageApplicationView/__tests__/utils.test.ts`
- Updated other tests the included refactoring touched.

### Manual testing :construction_worker_man:

- Application listing page:
  - The left sidebar should include a link to winter storage applications
  - Clicking the sidebar link should open the listing
  - The listing should display the information requested in the designs
  - The listing should be server-side paginated
- Single application page:
  - User be able to navigate to a single application's page by clicking on an application in the listing page
  - The page should contain all relevant information as specified in the designs
  - If no user is attached to the application, a selection list similar to the one present in berth applications' pages should be present
